vulcan/lib/asynchronous_scheduler.rb

```diff
@@ -45,18 +45,23 @@ def running?(build_target)
     def status(storage:, build_target:, step: nil)
       hash = build_target.cell_hash
 
-      requires_inputs = step&.ui_output_name && !build_target.is_buildable?(storage)
```

Comment:
This is the bug fix that ensures that we never report "complete" for a step just because it is built -- we ensure it also removed from the `@running` hash, which only ever occurs after we have attempted to schedule more dependent tasks.  Check the `ensure` block below.